### DISCUSSION TITLE
download all files

### DISCUSSION THREAD

**Discussion by waylennn** [minio/minio#17257] [Q&A] [answered]
Where in the minio source is a method with a url of 'minio/zip'? I want to download all the files in the folder

**Comment by klauspost** [comment, marked as answer]
There exists no such API feature in S3.

**Reply by waylennn** [reply]
> There exists no such API feature in S3.

I mean that the minio management platform can download the folder, its url path is "minio/zip", I want to go to the source code through this path to see how to achieve the download folder, but can not find the specific method function

**Comment by waylennn** [comment]
http://127.0.0.1:9000/minio/zip?token=eyJhbGciOiJIUzUxMiIsInR5cCI6IkpXVCJ9.eyJhY2Nlc3NLZXkiOiJjbG91ZGNoYWluIiwiZXhwIjoxNjg0NzQxMTI4LCJzdWIiOiJjbG91ZGNoYWluIn0.uBirXAPxWG20H4bOkCNNH1PM64ZAO7EE7Fz3XZ3d-X5tCul6mQgZhBh8OC8awHPb2vqGIAsMos-TkTyLk0660g

**Reply by waylennn** [reply]
The url is this, and I want to find the method function in the minio source code

**Reply by klauspost** [reply]
That was an old feature with a proprietary API. That has been removed a long time ago.

**Reply by waylennn** [reply]
> That was an old feature with a proprietary API. That has been removed a long time ago.


I just downloaded the latest code, and it's this url :  http://127.0.0.1:9090/api/v1/buckets/org1/objects/download?prefix=ZGF0YS8=.

**Reply by klauspost** [reply]
That is implemented in [console](https://github.com/minio/console), not the minio server.

Note it is calling console port, not the API port.

**Comment by MottiKumar** [comment]
why not try this it worked for me : 

`/usr/local/bin/mc share download --recursive --expire 1h "${BASE_MINIO_PATH}" 2>&1 | grep '^Share:' | awk '{print $2}' | xargs -I {} sh -c 'SIGNED_URL="{}"; RELATIVE_PATH=$(echo "$SIGNED_URL" | sed "s|http[s]*://[^/]*/||" | sed "s|\?.*$||"); mkdir -p "$(dirname "$RELATIVE_PATH")"; wget -O "$RELATIVE_PATH" "$SIGNED_URL"'`
